Fix for issue #927

tomas-pluskal opened the pull request:
Fixes issue #927. The most important change is that when the executor fails to retrieve a status of the job, it will assume the job is still running. I think it is a more meaningful approach than the current one.

Signed-off-by: Tomas Pluskal <plusik@gmail.com>

pditommaso commented on src/main/groovy/nextflow/executor/AbstractGridExecutor.groovy:
```diff
@@ -345,9 +345,10 @@ abstract class AbstractGridExecutor extends Executor {
             log.trace "[${name.toUpperCase()}] queue ${queue?"($queue) ":''}status > map does not contain jobId: `$jobId`"
-            return false
+            return true
         }
```
I don't agree on this change.

tomas-pluskal replied:
Reason?

pditommaso replied:
We are talking about that the batch scheduler you are using return inconsistent job status, therefore the flaw in the LFS installation you are using. Also LSF is a commercial product for which you are supposed to pay a commercial assistance. I don't why NF should take care of a bug in your system.

tomas-pluskal replied:
If you want nextflow to be successful and widely used, unfortunately it is necessary to take care of such edge cases. Plus, my main argument is that this is about the philosophy or approach that nextflow is taking here - instead of naively expecting that everything in the environment works as it should (which is obviously often not the case), it should take the approach that the environment is buggy and take the most conservative approach to recover from any possible failures. IMHO it would be a lot more useful workflow engine that way.

pditommaso commented on src/main/groovy/nextflow/executor/LsfExecutor.groovy:
```diff
@@ -142,7 +142,8 @@ class LsfExecutor extends AbstractGridExecutor {
-        final result = ['bjobs', '-w']
+        // use the `-a` option to include recently finished jobs
+        final result = ['bjobs', '-w', '-a']
 
```
Please update the relative tests, see below: https://github.com/nextflow-io/nextflow/blob/d4792d386dd9b900905254950445d8bae0ca8a75/src/test/groovy/nextflow/executor/LsfExecutorTest.groovy#L391-L429

tomas-pluskal commented:
Hi, I updated this PR to make the proposed change specific to the LSF executor by overriding the checkActiveStatus() method. This way the change won't affect the behavior of other executors, and thanks to the added 'bjobs -a' switch it should be a bulletproof solution for the LSF executor.

I also fixed the test as you suggested.

What do you think about this solution?

pditommaso commented:
This change completely alter the job monitoring logic. Is the cluster is not returning the job status it must be assumed terminated. NF cannot bit an unreliable behaviour of your cluster.
